### DISCUSSION TITLE
Admin UI customization

### DISCUSSION THREAD

**Discussion by danielgratzl** [keycloak/keycloak#17191] [Archive]
Hey there, I read in various places that the new way to extend the admin console is to fork the keycloak-ui repository.
What is not clear to me which branch should be used? 

There is a `release/20.0` branch, which I would have expected to be the correct branch to use with Keycloak 20.x but the POM file refers to Keycloak 19.0.3 (https://github.com/keycloak/keycloak-ui/blob/release/20.0/keycloak-theme/pom.xml)

while the POM in the `main` branch uses Keycloak 20.0.3 (https://github.com/keycloak/keycloak-ui/blob/main/keycloak-theme/pom.xml)

Building the UI from the release/20.0 branch and deploying with a 20.0.3 keycloak instance results in various startup errors

```
keycloak-themes-keycloak-1  | Caused by: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
keycloak-themes-keycloak-1  |   [error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: javax.enterprise.inject.spi.DeploymentException: Found 4 deployment problems: 
keycloak-themes-keycloak-1  | [1] Unsatisfied dependency for type org.keycloak.models.RealmModel and qualifiers [@Default]
keycloak-themes-keycloak-1  |   - java member: org.keycloak.admin.ui.rest.BruteForceUsersResource():realm
keycloak-themes-keycloak-1  |   - declared on CLASS bean [types=[org.keycloak.admin.ui.rest.BruteForceUsersResource, java.lang.Object], qualifiers=[@Default, @Any], target=org.keycloak.admin.ui.rest.BruteForceUsersResource]
keycloak-themes-keycloak-1  | [2] Unsatisfied dependency for type org.keycloak.services.resources.admin.permissions.AdminPermissionEvaluator and qualifiers [@Default]
keycloak-themes-keycloak-1  |   - java member: org.keycloak.admin.ui.rest.BruteForceUsersResource():auth
keycloak-themes-keycloak-1  |   - declared on CLASS bean [types=[org.keycloak.admin.ui.rest.BruteForceUsersResource, java.lang.Object], qualifiers=[@Default, @Any], target=org.keycloak.admin.ui.rest.BruteForceUsersResource]

```

**Comment by ssilvert** [comment]
@danielgratzl You can go to the tags page and download the source bundle for whatever release you want to base from:
https://github.com/keycloak/keycloak-ui/tags

**Reply by danielgratzl** [reply]
Thanks, I'm aware of that. 
Somehow for me this doesn't seem to work because as mentioned above the POM in the 20.0.x releases reference Keycloak 19.0.3 and  uses Keycloak 19.x sources for the extension of the REST Admin API which results in Startup errors.
